src/main/features/claimant-response/helpers/taskListBuilder.ts

```diff
@@ -27,21 +32,74 @@ export class TaskListBuilder {
     const tasks: TaskListItem[] = []
     const externalId: string = claim.externalId
 
-    if (claim.response.responseType === ResponseType.FULL_ADMISSION
-      || (claim.response.responseType === ResponseType.PART_ADMISSION && claim.response.paymentIntention !== undefined)) {
-      tasks.push(
-        new TaskListItem(
-          'View the defendant’s full response',
-          Paths.defendantsResponsePage.evaluateUri({ externalId: externalId }),
-          ViewDefendantResponseTask.isCompleted(draft.defendantResponseViewed)
-        )
+    tasks.push(
+      new TaskListItem(
+        'View the defendant’s full response',
+        Paths.defendantsResponsePage.evaluateUri({ externalId: externalId }),
+        ViewDefendantResponseTask.isCompleted(draft.defendantResponseViewed)
       )
-    }
+    )
 
     return new TaskList('Before you start', tasks)
   }
 
+  static buildStatesPaidHowYouWantToRespondSection (draft: DraftClaimantResponse, claim: Claim): TaskList {
+    const tasks: TaskListItem[] = []
+    const response: FullDefenceResponse | PartialAdmissionResponse = claim.response as FullDefenceResponse | PartialAdmissionResponse
+    const externalId: string = claim.externalId
+
+    if (response.responseType === ResponseType.FULL_DEFENCE) {
+      tasks.push(
+        new TaskListItem('Accept or reject their response',
+          Paths.settleClaimPage.evaluateUri({ externalId: externalId }),
+          ClaimSettledTask.isCompleted(draft)
+        ))
+    } else {
+      if (StatesPaidHelper.isAlreadyPaidLessThanAmount(claim)) {
+        tasks.push(
+          new TaskListItem(`Have you been paid the ${ NumberFormatter.formatMoney(response.amount) }?`,
+            Paths.partPaymentReceivedPage.evaluateUri({ externalId: externalId }),
+            PartPaymentReceivedTask.isCompleted(draft)
+          ))
+
+        if (draft.partPaymentReceived && draft.partPaymentReceived.received.option === YesNoOption.YES) {
+          tasks.push(
+            new TaskListItem(`Settle the claim for ${ NumberFormatter.formatMoney(response.amount) }?`,
+              Paths.settleClaimPage.evaluateUri({ externalId: externalId }),
+              ClaimSettledTask.isCompleted(draft)
+            ))
+        }
+      } else {
+        tasks.push(
+          new TaskListItem(`Have you been paid the full ${ NumberFormatter.formatMoney(claim.totalAmountTillDateOfIssue) }?`,
+            Paths.settleClaimPage.evaluateUri({ externalId: externalId }),
+            ClaimSettledTask.isCompleted(draft)
+          ))
+      }
+    }
+
+    if (claim.response.freeMediation === YesNoOption.YES) {
+      if ((draft.accepted && draft.accepted.accepted.option === YesNoOption.NO) ||
```

Comment:
nested if can be replaced with `&&`

Reply:
This was intentional, the condition became too complex and unreadable as 1 if statement